src/usage/search.md

```diff
@@ -1,83 +1,110 @@
+Set `hnsw` search scope to 40 in transaction:
+```sql
+SET LOCAL vectors.hnsw_ef_search=40;
+```
+
+Query options for `ivf`:
+
+| Option                   | Type    | Range            | Default | Description                               |
+| ------------------------ | ------- | ---------------- | ------- | ----------------------------------------- |
+| vectors.ivf_nprobe       | integer | `[1, 1_000_000]` | `10`    | Number of lists to scan.                  |
 
-Assuming you are using HNSW algorithm, you may want the following SQL to work:
+Query options for `hnsw`:
 
+| Option                   | Type    | Range          | Default | Description                               |
+| ------------------------ | ------- | -------------- | ------- | ----------------------------------------- |
+| vectors.hnsw_ef_search   | integer | `[1, 65535]`   | `100`   | Search scope of HNSW.                     |
+
+
+
+# Advanced usage
+
+Sometimes you expect the search to return the exact number of vectors equal to `LIMIT`, but it can't:
 ```sql
-SET vectors.hnsw_ef_search = 64;
-SELECT * FROM items ORDER BY embedding <-> '[3,2,1]' WHERE id % 2 = 0 LIMIT 64;
+SELECT COUNT(1) FROM (SELECT 1 FROM t WHERE (category_id = 1) ORDER BY val <-> '[1,1,1]' limit 10) t2;
+--- returns 1, much less than 10
 ```
+That is why we introduce search mode.
 
-In `basic` mode, you may only get `32` rows because the HNSW algorithm does search simply so the filter condition is ignored.
+## Search modes
 
-In `vbase` mode, the HNSW algorithm is guaranteed to return rows as many as you need, so you can always get correct behavior if your do take advantages of:
+There are two search modes: `basic` and `vbase`.
 
-* database transaction
-* deletions without `VACUUM`
-* `WHERE` clauses and very complex SQL statements
+Query options for search mode:
 
-You can enable `vbase` by a SQL statement `SET vectors.search_mode = vbase;`.
+| Option                   | Type    | Range              | Default   | Description                            |
+| ------------------------ | ------- | ------------------ | --------- | -------------------------------------- |
+| vectors.search_mode      | enum    | `"basic", "vbase"` | `"vbase"` | Search mode.                           |
 
-## Prefilter
+### `basic`
 
-If your queries include a `WHERE` clause, you can set set search mode to `vbase`. It's good and it even works on all conditions. `vbase` is a **postfilter** method: it pulls rows as many as you need, but it scans rows that you may not need. Since some rows will definitely be removed by the `WHERE` clause, we can skip scanning them, which will make the search faster. We call it **prefilter**.
+In this mode, the filter is applied after a batch of vectors vectors are returned.
+The number is determined by `vectors.hnsw_ef_search` for the `hnsw` algorithm, or `vectors.ivf_nprobe` for the `ivf` algorithm.
+Therefore you need to increase it until filtered vectors are sufficient.
```

Comment:
If it happens, ask users to use `vbase`. `basic` is for a tradition vector library usage, not for bothering users.